### PR TITLE
Use window.iframeLoaded promise

### PR DESCRIPTION
**Pull request by nmanu1**
Update iframe checks to use window.iframeLoaded promise. This is to avoid possible race conditions with the iframe (e.g. if the SDK adds the pagination component before iframe-resizer is spun up).

J=SLAP-1102
TEST=manual

Check default and iframe pages in test-site to make sure pagination scrolling works as expected for vertical map templates.

### CODE REVIEW

File: templates/vertical-map/script/pagination.hbs [hunk 1]
```diff
@@ -1,13 +1,14 @@
 ANSWERS.addComponent("Pagination", Object.assign({}, {
   container: "#js-answersPagination",
   onPaginate: (newPageNumber, oldPageNumber, totalPages) => {
-    if (window.parentIFrame) {
+    document.documentElement.scrollTop = 0;
+    document.body.scrollTop = 0; // Safari
```
Comment (yen-tt) [inline templates/vertical-map/script/pagination.hbs:5]: to clarify, this should be applied to both default and iframe page?

File: templates/common-partials/script/pagination.hbs [hunk 2]
```diff
@@ -1,13 +1,13 @@
 ANSWERS.addComponent("Pagination", Object.assign({}, {
   container: "#js-answersPagination",
   onPaginate: (newPageNumber, oldPageNumber, totalPages) => {
-    if (window.parentIFrame) {
+    document.documentElement.scrollTop = 0;
+    document.body.scrollTop = 0; // Safari
```
Comment (tmeyer2115) [inline templates/common-partials/script/pagination.hbs:5]: Does this scroll to the top of the iFrame or the top of the parent page?

File: templates/vertical-map/script/pagination.hbs (hunk 1 above)
Comment (nmanu1) [reply templates/vertical-map/script/pagination.hbs:5]: yes, this scrolls the page in the default case and scrolls the iFrame in an iframe page

File: script/core.hbs [hunk 3]
```diff
@@ -38,12 +38,12 @@
       ...token && { apiKey: token },
       querySource: window.isOverlay ? 'OVERLAY' : 'STANDARD',
       onStateChange: (objParams, stringParams, replaceHistory) => {
-        if ('parentIFrame' in window) {
+        window.iframeLoaded.then(() => {
```
Comment (oshi97) [inline script/core.hbs:41]: is there some chance of this being a memory leak, if window.iframeLoaded is never fulfilled? (i.e. we're on a non-iframe integration)

Comment (oshi97) [reply script/core.hbs:41]: despite onStateChange running for literally every SDK storage update I would think the memory still wouldn't matter, but hopefully there would be a relatively easy fix?

Comment (oshi97) [reply script/core.hbs:41]: note: Connor thought of this first I just parroted it

File: static/js/overlay/answers-frame/controllers/messagedirector.js [hunk 4]
```diff
@@ -73,6 +73,8 @@ export default class MessageDirector {
    * @param {IFrameMessage} message
    */
   _notifyParentFrame(message) {
-    window.parentIFrame && window.parentIFrame.sendMessage(message.toObject());
+    window.iframeLoaded.then(() => {
```
[the remark this answers is not in the document]
Comment (nmanu1) [reply static/js/overlay/answers-frame/controllers/messagedirector.js:76]: You're right. Looks like this is a different iframe, so I'll switch back to the previous check

**Pull request closed without merging**